test/integration/apiserver/admissionwebhook/invalid_managedFields_test.go

```diff
@@ -133,20 +133,21 @@ func TestMutatingWebhookResetsInvalidManagedFields(t *testing.T) {
 			lastErr = err
 			return false, nil
 		}
+		if warningWriter.WarningCount() != 1 {
+			lastErr = fmt.Errorf("expected one warning, got: %v", warningWriter.WarningCount())
+			return false, nil
+		}
+		if !strings.Contains(recordedWarnings.String(), expectedWarning) {
+			lastErr = fmt.Errorf("unexpected warning, expected: \n%v\n, got: \n%v",
+				expectedWarning, recordedWarnings.String())
+			return false, nil
+		}
+		lastErr = nil
 		return true, nil
-	}); err != nil {
+	}); err != nil || lastErr != nil {
 		t.Fatalf("failed to wait for apiserver handling webhook mutation: %v, last error: %v", err, lastErr)
 	}
-	if lastErr != nil {
-		t.Fatal(lastErr)
-	}
-	if warningWriter.WarningCount() != 1 {
-		t.Errorf("expected one warning, got: %v", warningWriter.WarningCount())
-	}
-	if !strings.Contains(recordedWarnings.String(), expectedWarning) {
-		t.Errorf("unexpected warning, expected: \n%v\n, got: \n%v",
-			expectedWarning, recordedWarnings.String())
-	}
+
 	recordedWarnings.Reset()
```

Comment:
shouldn't this move inside the loop as well?

Reply:
I don't think it has to, but it could yes.
Let me try and meanwhile I'll trigger the tests again.

Reply:
yeah, as written previously, the loop didn't actually ensure the webhook was called... if the webhook wasn't effective yet, the patch call would succeed with no warnings and fail the test

move the reset to the top of the loop with a `defer`, then lgtm

Reply:
just did that, wanted to wait for the result before I push but here it comes.